common/app/views/support/cleaner/AmpEmbedCleaner.scala

```diff
@@ -85,30 +92,86 @@ case class AmpEmbedCleaner(article: Article) extends HtmlCleaner {
     }
 
     def getTrackIdFromUrl(soundcloudUrl: String): Option[String] = {
-      val pattern = ".*soundcloud.com%2Ftracks%2F(\\d+).*".r
-      soundcloudUrl match {
-        case pattern(trackId) => Some(trackId)
+      val pattern = ".*api.soundcloud.com/tracks/(\\d+).*".r
+      URLDecoder.decode(soundcloudUrl,"UTF-8") match {
+        case pattern(trackId) => {
+          Some(trackId)}
         case _ => None
       }
     }
 
+    def getSoundCloudElement(document: Document, iframeElement: Element): Option[Element] = {
+      val trackId = AmpSoundcloud.getTrackIdFromUrl(iframeElement.attr("src"))
+      trackId.map(id => AmpSoundcloud.createElement(document, id))
+      /*  if (trackId.nonEmpty) {
+        Option(AmpSoundcloud.createElement(document, trackId.get))
+    } else {
+        None
+      }*/
+    }
+
+  }
+
+
+  //creates embeds for any elements of type "element-audio"
```

Comment:
Thanks Natalia, thats a very good point. I've relocated the "amp-iframe" bit to the relevant part of the method and stripped the rest out. The name does indeed provide sufficient information in this case. Thanks very much for pointing that out, I will keep it in mind for the future :-)